src/analyzer/analyzersilence.cpp

```diff
@@ -9,18 +9,31 @@ constexpr float kSilenceThreshold = 0.001;
 // TODO: Change the above line to:
 //constexpr float kSilenceThreshold = db2ratio(-60.0f);
 
-}  // anonymous namespace
+bool shouldUpdateMainCue(CuePosition cue) {
+    return cue.getSource() != Cue::MANUAL || cue.getPosition() == -1.0 || cue.getPosition() == 0.0;
+}
+
+bool hasIntroCueStart(const Cue& cue) {
```

Comment:
Is this function unrelated to into and outro?
Than we should omit intro in the function name.

Reply:
These position values are specific for the intro/outro cues. The encoding is not completely unambiguous and unfortunately scattered everywhere in the code.

The way how cue points are currently stored is suboptimal for representing both single points and partial or complete ranges. Those requirements were unknown at the time the data model was designed. This should be considered for a future version of the library.